packages/graphql-yoga/__tests__/http-extensions.spec.ts

```diff
@@ -613,20 +615,44 @@ describe('Result Extensions', () => {
         throw(status: $status)
       }
     `
-    for (const statusCodeStr in STATUS_CODES) {
-      const status = Number(statusCodeStr)
-      const res = await yoga.fetch('http://yoga/graphql', {
-        method: 'POST',
-        headers: {
-          'content-type': 'application/json',
-        },
-        body: JSON.stringify({
-          query,
-          variables: { status },
-        }),
-      })
-      expect(res.status).toBe(status)
-      expect(res.statusText).toBe(STATUS_CODES[status])
-    }
+
+    it('should set status text by status code with yoga.fetch(...)', async () => {
+      for (const statusCodeStr in STATUS_CODES) {
+        const status = Number(statusCodeStr)
+        const res = await yoga.fetch('http://yoga/graphql', {
+          method: 'POST',
+          headers: {
+            'content-type': 'application/json',
+          },
+          body: JSON.stringify({
+            query,
+            variables: { status },
+          }),
+        })
+        expect(res.status).toBe(status)
+        expect(res.statusText).toBe(STATUS_CODES[status])
+      }
+    })
+
+    it('should set status text by status code with yoga(...) returning res', async () => {
+      for (const statusCodeStr in STATUS_CODES) {
+        const status = Number(statusCodeStr)
+        const res = await yoga(
+          new Request('http://yoga/graphql', {
+            method: 'POST',
+            headers: {
+              'content-type': 'application/json',
+            },
+            body: JSON.stringify({
+              query,
+              variables: { status },
+            }),
+          }),
+          new Response(),
```

Comment:
Actually `yoga` handler doesn't work like that.
If you try to mimic `yoga(req, res)`, this is different.
There is no `Response` passed in Fetch API based servers. It is Node specific and parameters are `IncomingMessage` and `ServerResponse`.
In Fetch API based servers, the handler takes `Request` and returns `Response`.

Reply:
You can see all the combinations here;
https://github.com/ardatan/whatwg-node/blob/master/packages/server/src/types.ts#L29